instrumentation/opentelemetry-instrumentation-sklearn/src/opentelemetry/instrumentation/sklearn/__init__.py

```diff
@@ -0,0 +1,738 @@
+    ``transformer_list``. Instrumenting estimators like this is also
+    supported through the ``recurse_namedtuple_attribs`` argument. This
+    argument is a dictionary with classes as keys, and a list of attribute
+    names representing the namedtuple list(s). By default, the
+    ``recurse_namedtuple_attribs`` dictionary supports
+    ``Pipeline`` with ``steps``, and ``FeatureUnion`` with
+    ``transformer_list``.
+
+    Note that spans will not be generated for any child transformer whose
+    parent transformer has ``n_jobs`` parameter set to anything besides
+    ``None`` or ``1``.
+
+    Package instrumentation example:
+
+    .. code-block:: python
+
+        from opentelemetry.instrumentation.sklearn import SklearnInstrumentor
+
+        # instrument the sklearn library
+        SklearnInstrumentor().instrument()
+
+        # instrument additional libraries
+        packages = ["sklearn", "lightgbm", "xgboost"]
+        SklearnInstrumentor(packages=packages).instrument()
+
+
+    Model intrumentation example:
+
+    .. code-block:: python
+
+        from opentelemetry.instrumentation.sklearn import SklearnInstrumentor
+        from sklearn.datasets import load_iris
+        from sklearn.ensemble import RandomForestClassifier
+        from sklearn.model_selection import train_test_split
+        from sklearn.pipeline import Pipeline
+
+        X, y = load_iris(return_X_y=True)
+        X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+        model = Pipeline(
+            [
+                ("class", RandomForestClassifier(n_estimators=10)),
+            ]
+        )
+
+        model.fit(X_train, y_train)
+
+        SklearnInstrumentor().instrument_estimator(model)
+
+    Args:
+        methods (list): A list of method names on which to instrument a span.
+          This list of methods will be checked on all estimators in the model
+          hierarchy. Used in package and model instrumentation
+        recurse_attribs (dict): A dictionary of ``BaseEstimator``-derived
+          sklearn classes as keys, with values being a list of attributes. Each
+          attribute represents either an estimator or list of estimators on
+          which to also implement spans. An example is
+          ``RandomForestClassifier`` and its attribute ``estimators_``. Used
+          in model instrumentation only.
+        recurse_namedtuple_attribs (dict): A dictionary of ``BaseEstimator``-
+          derived sklearn types as keys, with values being a list of
+          attribute names. Each attribute represents a list of 2-tuples in
+          which the first element is the estimator name, and the second
+          element is the estimator. Defaults include sklearn's ``Pipeline``
+          and its attribute ``steps``, and the ``FeatureUnion`` and its
+          attribute ``transformer_list``. Used in model instrumentation only.
+        spanner: A function with signature (func, str) which
```

Comment:
Isn't the method signature `(func, BaseEstimator)`?

Reply:
Yes I'll fix this.